code/processes/tickerlogreplay.q

```diff
@@ -107,16 +107,19 @@ tablestoreplay:$[tablelist~enlist`all; tables[`.]; tablelist,()]
 .lg.o[`replayinit;"hdb directory is set to ",string hdbdir:hsym hdbdir];
 
 // get the list of log files to replay
+
 logstoreplay:$[not null tplogfile; 
-		[if[()~key hsym tplogfile; .err.ex[`replayinit;"specified tplogfile ",(string tplogfile)," does not exist";3]];
-		 enlist hsym tplogfile]; 
-		[.lg.o[`replayinit;"reading log files from directory ",string tplogdir];
-		 if[()~key hsym tplogdir; .err.ex[`replayinit;"specified tplogdir ",(string tplogdir)," does not exist";4]];
-		 raze ` sv' tplogdir,/:key tplogdir:hsym tplogdir]];
+	[if[()~key hsym tplogfile; .err.ex[`replayinit;"specified tplogfile ",(string tplogfile)," does not exist";3]];
+	$[segmentedmode; raze ` sv' tplogfile,/:key tplogfile:hsym tplogfile; enlist hsym tplogfile]]; 
+	[.lg.o[`replayinit;"reading log files from directory ",string tplogdir];
+	if[()~key hsym tplogdir; .err.ex[`replayinit;"specified tplogdir ",(string tplogdir)," does not exist";4]];
+	$[segmentedmode; `$raze (string pars),/:'string key each pars:hsym`$(string[tplogdir],"/"),/:string[key hsym tplogdir],'"/"; 
+    raze ` sv' tplogdir,/:key tplogdir:hsym tplogdir]]];
```

Comment:
I see what you are doing here, but I would probably leave this section as is, and "expand" the log directory when it comes to actually doing the replay, based on whether it is in segmented mode or not. So this bit stays as is (maybe need some minor changes) and then at replay time it expands the directory into the list of log files